### PR TITLE
[Winchus] Resubmission of Winchus keyboard

### PR DESCRIPTION
**Pull request by alex-larkin**
This is a resubmission of https://github.com/keymanapp/keyboards/pull/3023. Please view that pull request for more details, if needed.

### CODE REVIEW

**Comment by LornaSIL** [comment]
@alex-larkin Thanks for the PR! There are a few things that need adjusting:

- The LICENSE.md cannot have the first lines modified or our build system will croak. I assume the modification is just that you added a Spanish translation? If so, change the first lines to:
```
The MIT License (MIT)

Copyright © 2024 SIL Global
```

It needs that word `Copyright` in there even with the copyright symbol.

I *think* you can leave the bottom part as you have it, except the copyright line should be the same as above.

- Please delete the `README.htm` file in the root of `winchus`. That isn't necessary and I believe the relevant info is already in `README.md`.
- If you could remove the copyright line at the bottom of the readme.htm that would be great. Otherwise, make it `SIL Global` and not just `SIL`. 
- I'm not sure if the `welcome.htm` file is displaying correctly. It looks like it isn't. Can you double-check that's what you want?
- The php file is hard to parse. I can accept it, but since it's all on one line I'm not sure it's correct. Please don't include the table of contents in the php file. Our online build system automatically creates a table of contents so that would be a duplicate and messy.
- In the .kps 
   - **Details** tab can you make that `SIL Global`? That's what we are using these days. You could leave off the year in this case (we still need it in LICENSE.md) so we don't have to update that field again.
   - Again, in the **Details** tab can you add a description? That is what will show up on the home page, similar to what is under the keyboard name on this page: https://keyman.com/keyboards/sil_ipa

Otherwise it looks good!

**Comment by keyman-server** [comment]
Thank you for your pull request. You'll see a "build failed" message until the Keyman team has reviewed the pull request and manually initiated the build process.

Every change committed to this branch will become part of this pull request. When you have finished submitting files and are ready for the Keyman team to review this pull request, please post a "Ready for review" comment.

  PLEASE NOTE: team capacity for review is limited in December 2024 and will resume in January 2025. We appreciate your patience.

**Comment by alex-larkin** [comment]
Thanks so much Lorna for the quick response!

>The LICENSE.md cannot have the first lines modified or our build system will croak. I assume the modification is just that you added a Spanish translation? If so, change the first lines to:
>>The MIT License (MIT) 
>>
>>Copyright © 2024 SIL Global

Ok. I have modified the file and made the deviance from the standard MIT License more explicit. If there are still issues with this please let me know and I will remove the deviation. 

> Please delete the README.htm file in the root of winchus. 

Ok.

>If you could remove the copyright line at the bottom of the readme.htm that would be great. 

Ok.

>I'm not sure if the welcome.htm file is displaying correctly. It looks like it isn't. Can you double-check that's what you want?

My Windows Explorer preview window does mess up all accented vowels and other symbols not native to English orthography, but when I open the file in Chrome it does display as anticipated. 
image
What are you seeing? Should I worry about accented characters not displaying correctly on some platforms?

The heading highlighted in green explains that I'm still working on the document, but an easier-to-read version is available at the link (admittedly, for the moment the Google doc is full of number signs as I tried to convert it to another format without the carrots of `<kbd></kbd>` getting escaped.) (In PR #3023 @DavidLRowe suggested that I upload to the experimental folder while I work on polishing up the documentation.)

>The php file is hard to parse. I can accept it, but since it's all on one line I'm not sure it's correct. 

Ah yes, the PHP file. After I finish the .htm file I intend to obtain the PHP file by following these instructions: 
https://help.keyman.com/developer/keyboards/phphelpfile
Is it okay if I leave the PHP file as it is for now? Or, maybe I could leave it pretty much empty for now?

>Please don't include the table of contents in the php file. 

Ok, I will remove them from the PHP file. I have also removed it from the current draft (if I'm not mistaken. I also added a new line between every `>` and `<` in hopes to make it a little bit more accessible for the time being.)

Should I remove the TOC from the welcome file too? Does it get auto-generated there too?

>In the .kps Details tab can you make that SIL Global? That's what we are using these days. You could leave off the year in this case (we still need it in LICENSE.md) so we don't have to update that field again.

Ok, good catch. Thanks.

>Again, in the Details tab can you add a description? 

Thanks! Done.

This is the first time I have attempted to make corrections to a PR without canceling it. Please let me know if the changes didn't make it through okay, and what you think about welcome.htm and welcome.php.

**Comment by LornaSIL** [comment]
I deleted that one file.

Then I got to `LICENSE.md`. You may not modify the MIT license and remove any permission that the MIT license allows. Sorry. I thought you were just adding a translation.

This is the part where I wasn't sure I was seeing what you wanted me to see in `welcome.htm`
image

You can leave the TOC in the welcome.htm file.

As far as I can tell you've done everything I requested, so it's good to go except for the LICENSE modification.

**Comment by mcdurdin** [comment]
> Then I got to `LICENSE.md`. You may not modify the MIT license and remove any permission that the MIT license allows. Sorry. I thought you were just adding a translation.

Yes, the license cannot be modified, and the build will fail if there are any changes to the text -- only the copyright string may be modified. The MIT license has been chosen very deliberately for this repository, to grant the broadest possible freedoms we can to end users. All keyboards in the repository must be MIT licensed, because we cannot afford the complexity of managing variations (nor the legal review of variations to licenses). You can of course distribute a keyboard outside the repository with any license you please, but this is a non-negotiable condition of submission to the keyboards repository.

Secondly, the license check will fail because of the additional text. If you want to include a translation of the license, include it in a separate file, e.g. LICENSE.es.md. You should add a sentence at the top of the translated file noting that the translation has not been legally checked and is for reference only, and LICENSE.md forms the legal license.

Some additional discussion here: https://opensource.stackexchange.com/questions/6077/license-details-in-languages-other-than-english-american

**Comment by alex-larkin** [comment]
Thanks Lorna! 

I found a tutorial on how to delete files but you deleted and committed about 30 seconds before me 😂.

And yes, the welcome files are still a mess. I am manually coding everything because I haven't found another way to do it that doesn't fill the file with a bunch of HTML gunk.

I totally understand Keyman's issue around licensing. I have brought it into compliance as indicated by both of you.  I pasted in the [English ](https://en.wikipedia.org/wiki/MIT_License)and [Spanish ](https://es.wikipedia.org/wiki/Licencia_MIT)licenses as given on Wikipedia. 

I noticed that [.keyboard_info 2.0 Specification](https://help.keyman.com/developer/cloud/keyboard_info/current-version/) states, 
>release/ and experimental/ keyboards must be "mit".

indicating that new repository keyboards do need to have the MIT license. 

[Preliminary Step: Is Your Keyboard Ready?](https://help.keyman.com/developer/keyboards/advanced/step-0) states the following:
>LICENSE.md
>>the open source license for the keyboard. This file should include a copyright message, including a date or range of dates encompassing the timeframe in which the keyboard was first created until the most recent version...

Maybe it could be clarified to indicate the MIT license is required for all submissions to the repository?

Thanks so much for all of your feedback! It's really helping. 

Is everything ready, or do I need to edit something else? 

Thanks again!

**Comment by mcdurdin** [comment]
> Maybe it could be clarified to indicate the MIT license is required for all submissions to the repository?

https://github.com/keymanapp/help.keyman.com/pull/1735

**Review by LornaSIL — APPROVED** [review]
LGTM. FYI: In the future, we do prefer file names without spaces.

**Comment by alex-larkin** [comment]
Thanks so much Lorna and Marc. My first-ever merged PR! Whoo hoo! 🥳

**Comment by LornaSIL** [comment]
Congratulations! ⚡
